src/modules/referrals/client.spec.ts

```diff
@@ -59,3 +59,81 @@ describe('when the API returns a 503 error', () => {
     await expect(client.getReferralProgress()).rejects.toThrow('HTTP error! status: 503')
   })
 })
+
+describe('when setting referral email', () => {
+  let testEmail: string
+  beforeEach(() => {
+    testEmail = 'test@example.com'
+  })
+
+  describe('and the request is successful', () => {
+    beforeEach(() => {
+      mockFetch.mockResolvedValue(undefined)
+    })
+
+    it('should make a POST request to the referral email endpoint', async () => {
+      await client.setReferralEmail(testEmail)
+
+      expect(mockFetch).toHaveBeenCalledWith('/v1/referral-email', {
+        method: 'POST',
+        body: JSON.stringify({ email: testEmail })
+      })
+    })
+
+    it('should return void when successful', async () => {
+      const response = await client.setReferralEmail(testEmail)
+
+      expect(response).toBeUndefined()
+    })
+  })
+
+  describe('and the request fails with 400 (InvalidRequestError)', () => {
```

Comment:
As checking these all of these cases kind of implies checking that the `BaseClient` works as expected, we could leave only one of the tests just to check a failed request instead of having many of them, what do you think about it?